In [1]:
import pandas as pd

In [2]:
dataset = pd.read_csv("Placement.csv")

In [3]:
dataset

,sl_no,gender,ssc_p,ssc_b,hsc_p,hsc_b,hsc_s,degree_p,degree_t,workex,etest_p,specialisation,mba_p,status,salary
0,1,M,67.00,Others,91.00,Others,Commerce,58.00,Sci&Tech,No,55.0,Mkt&HR,58.80,Placed,270000.0
1,2,M,79.33,Central,78.33,Others,Science,77.48,Sci&Tech,Yes,86.5,Mkt&Fin,66.28,Placed,200000.0
2,3,M,65.00,Central,68.00,Central,Arts,64.00,Comm&Mgmt,No,75.0,Mkt&Fin,57.80,Placed,250000.0
3,4,M,56.00,Central,52.00,Central,Science,52.00,Sci&Tech,No,66.0,Mkt&HR,59.43,Not Placed,NaN
4,5,M,85.80,Central,73.60,Central,Commerce,73.30,Comm&Mgmt,No,96.8,Mkt&Fin,55.50,Placed,425000.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
210,211,M,80.60,Others,82.00,Others,Commerce,77.60,Comm&Mgmt,No,91.0,Mkt&Fin,74.49,Placed,400000.0
211,212,M,58.00,Others,60.00,Others,Science,72.00,Sci&Tech,No,74.0,Mkt&Fin,53.62,Placed,275000.0
212,213,M,67.00,Others,67.00,Others,Commerce,73.00,Comm&Mgmt,Yes,59.0,Mkt&Fin,69.72,Placed,295000.0
213,214,F,74.00,Others,66.00,Others,Commerce,58.00,Comm&Mgmt,No,70.0,Mkt&HR,60.23,Placed,204000.0


# Central Tendency

In [6]:
dataset["ssc_p"].mean()

np.float64(67.30339534883721)

In [7]:
dataset["ssc_p"].median()

np.float64(67.0)

In [10]:
dataset["ssc_p"].mode()

0    62.0
Name: ssc_p, dtype: float64

In [11]:
dataset["ssc_p"].mode()[0]

np.float64(62.0)

In [18]:
dataset.describe()

,sl_no,ssc_p,hsc_p,degree_p,etest_p,mba_p,salary
count,215.000000,215.000000,215.000000,215.000000,215.000000,215.000000,148.000000
mean,108.000000,67.303395,66.333163,66.370186,72.100558,62.278186,288655.405405
std,62.209324,10.827205,10.897509,7.358743,13.275956,5.833385,93457.452420
min,1.000000,40.890000,37.000000,50.000000,50.000000,51.210000,200000.000000
25%,54.500000,60.600000,60.900000,61.000000,60.000000,57.945000,240000.000000
50%,108.000000,67.000000,65.000000,66.000000,71.000000,62.000000,265000.000000
75%,161.500000,75.700000,73.000000,72.000000,83.500000,66.255000,300000.000000
max,215.000000,89.400000,97.700000,91.000000,98.000000,77.890000,940000.000000


In [22]:
descriptive = pd.DataFrame()

In [23]:
descriptive

""


In [24]:
descriptive = pd.DataFrame(index=["Mean","Median","Mode"])

In [25]:
descriptive

""
Mean
Median
Mode


In [26]:
from univariate import Univariate

In [27]:
Quan,Qual=Univariate.QuanQual(dataset)

In [28]:
Quan

['sl_no', 'ssc_p', 'hsc_p', 'degree_p', 'etest_p', 'mba_p', 'salary']

In [29]:
descriptive = pd.DataFrame(index=["Mean","Median","Mode"], columns=Quan)

In [30]:
descriptive

,sl_no,ssc_p,hsc_p,degree_p,etest_p,mba_p,salary
Mean,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Median,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Mode,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [31]:
descriptive["hsc_p"]["Median"]=0

C:\Users\pripo\AppData\Local\Temp\ipykernel_18752\2333899791.py:1: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!
You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.
A typical example is when you are setting values in a column of a DataFrame, like:

df["col"][row_indexer] = value

Use `df.loc[row_indexer, "col"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

  descriptive["hsc_p"]["Median"]=0


In [32]:
descriptive

,sl_no,ssc_p,hsc_p,degree_p,etest_p,mba_p,salary
Mean,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Median,NaN,NaN,0,NaN,NaN,NaN,NaN
Mode,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [37]:
for col in Quan:
    descriptive[col]["Mean"]=dataset[col].mean()
    descriptive[col]["Median"]=dataset[col].median()
    descriptive[col]["Mode"]=dataset[col].mode()[0]
    

C:\Users\pripo\AppData\Local\Temp\ipykernel_18752\330314331.py:2: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!
You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.
A typical example is when you are setting values in a column of a DataFrame, like:

df["col"][row_indexer] = value

Use `df.loc[row_indexer, "col"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

  descriptive[col]["Mean"]=dataset[col].mean()
C:\Users\pripo\AppData\Local\Temp\ipykernel_18752\330314331.py:3: FutureWarning: Cha

In [38]:
descriptive

,sl_no,ssc_p,hsc_p,degree_p,etest_p,mba_p,salary
Mean,108.0,67.303395,66.333163,66.370186,72.100558,62.278186,288655.405405
Median,108.0,67.0,65.0,66.0,71.0,62.0,265000.0
Mode,1,62.0,63.0,65.0,60.0,56.7,300000.0


# Summary ssc_p
1. Mean value ssc_p: Out of Placement dataset the mean value of ssc_p mark is 67.3
2. Median value ssc_p: Out of Placement dataset the median value of ssc_p mark is 67. 
3. Mode value ssc_p: Out of Placement dataset the mode value of ssc_p mark is 62. Most of students got 62

# Summary hsc_p
1. Mean value hsc_p: Out of Placement dataset the mean value of hsc_p mark is 66.3
2. Median value hsc_p: Out of Placement dataset the median value of hsc_p mark is 65. 
3. Mode value hsc_p: Out of Placement dataset the mode value of hsc_p mark is 63. Most of students got 63

# Summary degree_p
1. Mean value degree_p: Out of Placement dataset the mean value of degree_p mark is 66.3
2. Median value degree_p: Out of Placement dataset the median value of degeree_p mark is 66. 
3. Mode value degree_p: Out of Placement dataset the mode value of degree_p mark is 65. Most of students got 65

# Summary etest_p
1. Mean value etest_p: Out of Placement dataset the mean value of etest_p mark is 72.1
2. Median value etest_p: Out of Placement dataset the median value of etest_p mark is 71. 
3. Mode value etest_p: Out of Placement dataset the mode value of etest_p mark is 60. Most of students got 60

# Summary mba_p
1. Mean value mba_p: Out of Placement dataset the mean value of mba_p mark is 62.2
2. Median value mba_p: Out of Placement dataset the median value of mba_p mark is 62. 
3. Mode value mba_p: Out of Placement dataset the mode value of mba_p mark is 56.7. Most of students got 56.7

# Summary salary
1. Mean value salary: Out of Placement dataset the mean value of salary mark is 288655.4
2. Median value salary: Out of Placement dataset the median value of salary mark is 265000.0. 
3. Mode value salary: Out of Placement dataset the mode value of salary mark is 300000.0. Most of students got 300000.0